src/lib/config/ModelConfig.cc

```diff
@@ -0,0 +1,997 @@
+      std::string key = std::string(chld.key().data(), chld.key().size());
+      ref = configTree_.ref(id).append_child() << ryml::key(key);
+      // Set any appropriate ryml::NodeRef types
+      if (chld.is_map()) {
+        ref |= ryml::MAP;
+      }
+      if (chld.is_seq()) {
+        ref |= ryml::SEQ;
+      }
+    } else {
+      ref = configTree_.ref(id)[chld.key()];
+    }
+    if (chld.is_map()) {
+      // If the config option had children, iterate through them.
+      recursiveAdd(chld, ref.id());
+    } else if (chld.is_seq()) {
+      // If the config option is a sequence, then add the sequence of values
+      // held within the config option (its children) as children to the current
+      // ryml::Tree node identified by `id`
+      ref.clear_children();
+      for (size_t entry = 0; entry < chld.num_children(); entry++) {
+        ref.append_child();
+        ref[entry] << chld[entry].val();
+      }
+    } else {
+      // If the config option is neither a map nor a sequence, simply add its
+      // value to the ryml::Tree node reference
+      ref << chld.val();
+    }
+  }
+}
+
+void ModelConfig::setExpectations(bool isDefault) {
+  // Reset expectations
+  expectations_ = {};
+
+  // Core
+  expectations_.addChild(ExpectationNode::createExpectation("Core"));
+
+  if (ISA_ == ISA::AArch64)
+    expectations_["Core"].addChild(
+        ExpectationNode::createExpectation<std::string>("AArch64", "ISA"));
+  else if (ISA_ == ISA::RV64)
+    expectations_["Core"].addChild(
+        ExpectationNode::createExpectation<std::string>("rv64", "ISA"));
+  expectations_["Core"]["ISA"].setValueSet(
+      std::vector<std::string>{"AArch64", "rv64"});
+
+  // Early check on [Core][ISA] as its value is needed to inform the
+  // expectations of other config options
+  if (!isDefault) {
+    ValidationResult result = expectations_["Core"]["ISA"].validateConfigNode(
+        configTree_["Core"]["ISA"]);
+    std::string ISA;
+    configTree_["Core"]["ISA"] >> ISA;
+    if (result.errored) {
+      std::cerr << "[SimEng:ModelConfig] Invalid ISA value of \"" << ISA
+                << "\" passed in config file due to \"" << result.message
+                << "\" error. Cannot continue with config validation. Exiting."
+                << std::endl;
+      exit(1);
+    }
+    // Set ISA_
+    if (ISA == "AArch64") {
+      ISA_ = ISA::AArch64;
+    } else if ("rv64") {
+      ISA_ = ISA::RV64;
+    }
+  }
+  createGroupMapping();
+
+  expectations_["Core"].addChild(
+      ExpectationNode::createExpectation<std::string>("emulation",
+                                                      "Simulation-Mode"));
+  expectations_["Core"]["Simulation-Mode"].setValueSet(
+      std::vector<std::string>{"emulation", "inorderpipelined", "outoforder"});
+
+  expectations_["Core"].addChild(
+      ExpectationNode::createExpectation<float>(1.f, "Clock-Frequency"));
+  expectations_["Core"]["Clock-Frequency"].setValueBounds(0.f, 10.f);
```

Comment:
Proposing a change to `Clock-Frequency-GHz` for additional clarity

Reply:
Somewhat agree with this but where does it end regarding the other config options? Should we also have stuff like `Load-Bandwidth-Bytes`

Reply:
I personally don't object to being more verbose with all config options, but specifically about `Clock-Frequency`; given `Timer-Frequency` doesn't abide by the same units I feel that clarification could be useful whereas all bandwidths are given in bytes (and are generally the accepted standard unit for those metrics, but so it GHz and clock frequency....)